### PR TITLE
UN-180 and UN-181 Category model and crud

### PR DESCRIPTION
**Pull request by coledykstra**
UN-180 Add Category model and migration
UN-181 Add basic Category crud controller

### CODE REVIEW

File: app/Http/Controllers/v1/CategoryController.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+class CategoryController extends Controller
+{
+    public function index()
+    {
+        $categories = Category::all();
+
+        return response()->json($categories, Response::HTTP_OK);
```
Comment (PThorpe92) [inline app/Http/Controllers/v1/CategoryController.php:16]: We return API responses with Laravel's `Resource` type. You can do `php artisan make:resource CategoryResource`

You can look at Chris' latest PR for reference, but you are essentially defining the structure of exactly what you want to return. [here](https://github.com/chrissantillan/UnlockEdv2/blob/17bbac8fb3ecad79bbbab65ab0468443e3148aa7/app/Http/Resources/ProviderPlatformResource.php)

Then you can return things from your controllers by doing `CategoriesResource::collection($categories)`

Also you can do `Category::all()->paginate(10);`

File: app/Http/Controllers/v1/CategoryController.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        return response()->json($category, Response::HTTP_OK);
+    }
+
+    public function store(Request $request)
+    {
+        $validator = validator($request->all(), [
+            'name' => 'required|string|max:255',
+            // Add other validation rules for your fields
```
Comment (PThorpe92) [inline app/Http/Controllers/v1/CategoryController.php:34]: // Add other validation rules for your fields ;)

File: database/seeders/CategorySeeder.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+use Illuminate\Database\Seeder;
+
+class CategorySeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     */
+    public function run(): void
+    {
+        //
```
Comment (PThorpe92) [inline database/seeders/CategorySeeder.php:14]: This should run the seeder, you can look at others in the code base for reference

File: app/Http/Controllers/v1/CategoryController.php (hunk 2 above)
Comment (PThorpe92) [reply app/Http/Controllers/v1/CategoryController.php:34]: You can reference other Request types in the code base, so validation can be called on the requests with the
`$request->validated()` method. 
[reference](https://github.com/chrissantillan/UnlockEdv2/blob/17bbac8fb3ecad79bbbab65ab0468443e3148aa7/app/Http/Requests/StoreProviderPlatformRequest.php)

can also be generated with the `php artisan make:request CategoryRequest`

**Review by PThorpe92 — CHANGES_REQUESTED** [review]

**Comment by coledykstra** [comment]
I did the `git commit -m "fix: resolved api returns Category Resources and validator checks 3 field"` but i am lost what to do next.

**Review by PThorpe92 — APPROVED** [review]

**Review by nokierae — APPROVED** [review]

**Pull request merged**